Plug-Ins/NovaTools/System Editor/SystemMapView.swift

```diff
@@ -0,0 +1,307 @@
+        let path = NSBezierPath()
+        NSColor(deviceRed: 0.5, green: 0.5, blue: 0.5, alpha: 0.4).setStroke()
+        path.move(to: NSPoint(x: 0, y: frame.size.height / 2 - 0.5))
+        path.line(to: NSPoint(x: frame.size.width, y: frame.size.height / 2 - 0.5))
+        path.move(to: NSPoint(x: frame.size.width / 2 - 0.5, y: 0))
+        path.line(to: NSPoint(x: frame.size.width / 2 - 0.5, y: frame.size.height))
+        path.stroke()
+    }
+
+    // MARK: - Pan and zoom
+
+    // Rescale all zoom levels so the 5k x 5k view can show the whole system (30k x 30k) even on max zoom
+    private let zoomLevels: [Double] = [8/19, 9/16, 3/4, 1, 4/3, 16/10]
+    func scaleFactor(forLevel level: Int) -> CGFloat {
+        return zoomLevels[level] * (5 / 30); // Scale the view to no more than 5k pixels across to avoid issues
+    }
+    private var zoomLevel = 0 {
+        didSet {
+            if zoomLevel != oldValue {
+                let oldPos = NSPoint(x: (enclosingScrollView?.documentVisibleRect.midX ?? 0) / frame.size.width, y: (enclosingScrollView?.documentVisibleRect.midY ?? 0) / frame.size.height)
+                let scaleFactor = scaleFactor(forLevel: zoomLevel)
+                // EVN wraps the player around at ±15000 from the origin, so a stellar could conceivably be placed anywhere within
+                // that range
+                let calculatedSize = NSSize(width: Int(30000 * scaleFactor), height: Int(30000 * scaleFactor))
+                frame.size = calculatedSize
+
+                transform = AffineTransform(translationByX: frame.midX, byY: frame.midY)
+                transform.scale(scaleFactor)
+
+                self.transformSubviews()
+
+                if let newVisibleRect = enclosingScrollView?.documentVisibleRect {
+                    let scrollOrigin = NSPoint(x: oldPos.x * frame.size.width - newVisibleRect.width / 2, y: oldPos.y * frame.size.height - newVisibleRect.height / 2)
+                    self.scroll(scrollOrigin)
+                }
+                enclosingScrollView?.flashScrollers()
+            }
+        }
+    }
+
+    @IBAction func zoomIn(_ sender: Any) {
+        zoomLevel = min(zoomLevel + 1, zoomLevels.endIndex - 1)
+    }
+
+    @IBAction func zoomOut(_ sender: Any) {
+        zoomLevel = max(zoomLevel - 1, zoomLevels.startIndex)
+    }
+
+    // Drag background to pan
+    override func mouseDragged(with event: NSEvent) {
+        if let clipView = superview as? NSClipView {
+            var origin = clipView.bounds.origin
+            origin.x -= event.deltaX
+            origin.y -= event.deltaY
+            self.scroll(origin)
+        }
+    }
+
+    // MARK: - Selection
+
+    private(set) var selectedStellars: [StellarView] = []
+
+    override func selectAll(_ sender: Any?) {
+        controller.stellarTable.selectAll(self)
+    }
+
+    // Click background to deselect (if not holding shift or command and not dragging)
+    // Double click to create system
```

Comment:
```suggestion
    // Double click to create stellar
```
(Also in a few more places below)